src/adam7.rs

```diff
@@ -129,21 +130,25 @@ fn expand_adam7_bits(width: usize, info: &InterlaceInfo, bits_pp: usize) -> Step
 
     // the equivalent line number in progressive scan
     let prog_line = line_mul * line_no + line_off;
-    // line width is rounded up to the next byte
-    let line_width = (width * bits_pp + 7) & !7;
-    let line_start = prog_line * line_width;
-    let start = line_start + (samp_off * bits_pp);
-    let stop = line_start + (width * bits_pp);
+    let line_start = prog_line * row_stride_in_bytes * 8;
```

Comment:
What's to stop overflow here if the caller passes in an unreasonable `row_stride_in_bytes`? Perhaps you should use checked maths here.

Reply:
> What's to stop overflow here if the caller passes in an unreasonable `row_stride_in_bytes`? Perhaps you should use checked maths here.

You are correct to observe that nothing will stop an overflow if a big `row_stride_in_bytes` is passed here.  This seems okay, because this will not lead to memory safety issues (i.e. in safe Rust code, the runtime checks will prevent reaching out-of-bounds memory).  So, I think that checked math is not necessary here.  I think that this can be seen as a similar decision to not check the size of `interlaced_row` or `img` in `fn expand_interlaced_row` - I think this is also ok.